drivers/soundwire/intel.c

```diff
@@ -408,6 +408,39 @@ static void intel_shim_wake(struct sdw_intel *sdw, bool wake_enable)
 	}
 }
 
+void sdw_intel_jack_detect(struct sdw_intel_ctx *ctx)
+{
+	struct sdw_intel_link_res *link;
+	struct sdw_master_device *md;
+	struct sdw_slave *slave;
+	struct sdw_bus *bus;
+
+	list_for_each_entry(link, &ctx->link_list, list) {
+		struct sdw_intel *sdw;
+		u16 wake_sts;
+
+		md = link->md;
+		bus = &link->cdns->bus;
+		sdw = cdns_to_intel(link->cdns);
+		wake_sts = intel_readw(sdw->link_res->shim, SDW_SHIM_WAKESTS);
+
+		if (!(wake_sts & BIT(sdw->instance)))
+			continue;
+
+		/*
+		 * wake up master and slave so that slave can notify master
+		 * the jack event and let codec driver check jack status
+		 */
+		list_for_each_entry(slave, &bus->slaves, node) {
+			if (slave->prop.wake_capable) {
+				pm_runtime_get(&slave->dev);
+				pm_runtime_put(&slave->dev);
```

Comment:
sorry, but this looks like a horrible hack.
what are you trying to achieve with this?

I suspect the idea is that we don't know which Slave device generated the wake, so we request all Slaves to become active and each driver will then double-check the interrupt source that caused the wake.

but isn't this what pm_runtime_request_resume() is all about

````
 int pm_request_resume(struct device *dev);
    - submit a request to execute the subsystem-level resume callback for the
      device (the request is represented by a work item in pm_wq); returns 0 on
      success, 1 if the device's runtime PM status was already 'active', or
      error code if the request hasn't been queued up
````

Reply:
> sorry, but this looks like a horrible hack.
> what are you trying to achieve with this?
> 
> I suspect the idea is that we don't know which Slave device generated the wake, so we request all Slaves to become active and each driver will then double-check the interrupt source that caused the wake.

yes, this is my idea. we only know which master trigger a interrupt, but don't know which slave.
I will refine it in another way.